presto-session-property-managers/src/main/java/io/prestosql/plugin/session/SessionMatchSpec.java

```diff
@@ -120,4 +127,44 @@ public SessionMatchSpec(
     {
         return sessionProperties;
     }
+
+    public static class Mapper
+            implements RowMapper<SessionMatchSpec>
+    {
+        @Override
+        public SessionMatchSpec map(ResultSet resultSet, StatementContext context) throws SQLException
+        {
+            Map<String, String> sessionProperties = getProperties(
+                    Optional.ofNullable(resultSet.getString("session_property_names")),
+                    Optional.ofNullable(resultSet.getString("session_property_values")));
+
+            return new SessionMatchSpec(
+                    Optional.ofNullable(resultSet.getString("user_regex")).map(Pattern::compile),
+                    Optional.ofNullable(resultSet.getString("source_regex")).map(Pattern::compile),
+                    Optional.ofNullable(resultSet.getString("client_tags")).map(tag -> Splitter.on(",").splitToList(tag)),
+                    Optional.ofNullable(resultSet.getString("query_type")),
+                    Optional.ofNullable(resultSet.getString("group_regex")).map(Pattern::compile),
+                    sessionProperties);
+        }
+
+        private static Map<String, String> getProperties(Optional<String> names, Optional<String> values)
+        {
+            if (!names.isPresent()) {
+                return ImmutableMap.of();
+            }
+
+            checkArgument(values.isPresent() == true);
```

Comment:
Also, can `values` be present when `names` are not?

Reply:
@findepi sorry I missed this one. It is usually not possible that one is present but the other is not since these columns are generated from `GROUP_CONCAT`. but since `getString` can return a NULL, for sanity of this method, we check whether values are present or not before getting the field out of the optional.